### PR TITLE
Handle legacy root contains in clearContainer

### PR DESCRIPTION
**Pull request by trueadm**
When debugging an internal issue, I noticed that we have an issue where we clear down a root that has existing content. This particular fix checks in `clearContainer` if the container we're trying to clear has a legacy container and if it does, checks to see if it matches the root we expected.

### CODE REVIEW

File: packages/react-reconciler/src/ReactFiberCompleteWork.old.js [hunk 1]
```diff
@@ -679,7 +679,7 @@ function completeWork(
           // If we hydrated, then we'll need to schedule an update for
           // the commit side-effects on the root.
           markUpdate(workInProgress);
-        } else if (!fiberRoot.hydrate) {
+        } else if (!fiberRoot.hydrate && workInProgress.child !== null) {
```
Comment (bvaughn) [inline packages/react-reconciler/src/ReactFiberCompleteWork.old.js:682]: Hm...so the way this check is supposed to work is this:

If the current Fiber didn't render any children (`current.child === null`) then that means we are either mounting (in which case the container should be clear, or we should clear it) or updating (in which case, if `current.child` is null, our previous render didn't render any children- so clearing the container should be a noop anyway).

The work in progress fiber having a child just means "we are about to add children to this container".

**Comment by gaearon** [comment]
Btw don't forget to update both files.

File: packages/react-reconciler/src/ReactFiberCompleteWork.old.js (hunk 1 above)
Comment (trueadm) [reply packages/react-reconciler/src/ReactFiberCompleteWork.old.js:682]: Shouldn't the work in progress have children? If it doesn't, aren't we rendering nothing over previously having nothing?

**Comment by trueadm** [comment]
This was more for a reference to the debugging work with @bvaughn. I don't expect to merge it.

File: packages/react-reconciler/src/ReactFiberCompleteWork.old.js (hunk 1 above)
Comment (bvaughn) [reply packages/react-reconciler/src/ReactFiberCompleteWork.old.js:682]: Whether the work in progress has children is not relevant to the specific question of whether the container has non-React contents in it. That's what we're trying to clear here. If we know that React didn't add things to this container the last time we committed (or if we are mounting) then we clear the container to make sure there's nothing unexpected in it.

**Comment by trueadm** [comment]
I've updated this PR with another possible fix.

**Comment by bvaughn** [comment]
This PR fixes the regression we're seeing internally. I think there might be a better place to check for this though. Let me post a potential alternate PR here shortly...

**Pull request closed without merging**